src/components/HeartbeatBar.vue

```diff
@@ -111,6 +98,47 @@ export default {
             return placeholders.concat(this.beatList.slice(start));
         },
 
+        values() {
+            if (!this.shortBeatList || !this.shortBeatList.length) {
+                return [];
+            }
+
+            const valueObj = {};
```

Comment:
[`Map`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Map)'s are generally prefered for storing key-value pairs in JS, afaik